templates/CRM/Price/Page/LineItem.tpl

```diff
@@ -67,7 +67,10 @@
       <td class="right">{$line.line_total|crmMoney:$currency}</td>
       {if $line.tax_rate != "" || $line.tax_amount != ""}
         <td class="right">{$taxTerm} ({$line.tax_rate}%)</td>
-        <td class="right">{$line.tax_amount|crmMoney:$currency}</td>
+        <td class="right">
+          {math equation="(x * y) / 100" x=$line.line_total y=$line.tax_rate assign=tax_value}
```

Comment:
Please do not perform rounding calculations at the TPL layer. This should just be a variable exposed to the TPL and calculated elsewhere. The rounding function itself should be dependent on the localisation set by the system.

Reply:
@agileware-justin - the crmMoney function uses php layer functionality - so it IS using the localisation set by the system as I understand it.

Reply:
From what I recall the issue is that the php lines would need to be wrangled in every form that assigns to this page in order to get it consistently right. 

It was in the BAO layer - which was causing the wrong info to be assigned to the DB, so that was wrong. It looks like the PR removing that rounding got merged which fixed one rounding presentation error & the data storage error - but might cause other errors.

Reply:
`{math equation="(x * y) / 100" x=$line.line_total y=$line.tax_rate assign=tax_value}`

So my question relates to this line, why do that here? Isn't this the pattern we are trying to stamp out?

Reply:
yeah - I don't know to be honest. I'm on the fence as to whether it's actually good to do things in the tpl / presentation layer, or just expedient.

What I think we want to stamp out in the tpl layer is things that are given meanings they don't intrinsically have - this is not a great example but it's from the same file

```
        {if $pricesetFieldsCount}
          <th class="right">{ts}Total Participants{/ts}</th>
        {/if}
```

So ``$pricesetFieldsCount``` 

in fact means ```$isMultipleParticipants ``` or something like that. Commonly we see  things like 
``{if $unrelatedVariableHistoricallySetIntheSameLoop}``... which is the more extreme example of that

Reply:
I would agree with Justin on this, I would have thought that the original $tax_amount would already be correct right or is there something I am missing?

Reply:
I should also note the revert rounding in BAO patch was merged more quickly than I expected - if it causes problems we should revert it before 5.28 goes out & head back to the drawing board.

Reply:
@seamuslee001 so it might be accurate now the rounding is out of the BAO - I became more OK with that approach as I dug. I haven't re-tested this since you merged the BAO rounding one .

I'm not sure what other fall out that will have though - as you can see in both the before & after screenshots there is a rounding issue before hand& one after  (per comments)

At this stage I'm much more interested in figuring out if we can get the money formatting function working the math part since the math issue probably needs re-defining now. as it has surely moved :-)